crates/rolldown_common/src/chunk/mod.rs

```diff
@@ -168,8 +175,16 @@ impl Chunk {
         hash
       }
     });
+    let chunk_name = if options.preserve_modules
+      && matches!(self.kind, ChunkKind::EntryPoint { is_user_defined, .. } if !is_user_defined)
```

Comment:
I did some hack because our `is_user_defined` seems to have different logic about chunk name generation, I will try to normalize it in the next pr.